src/components/Terminal.tsx

```diff
@@ -200,17 +213,18 @@ class Terminal extends React.Component<ThemedTerminalProps, {}> {
 
 	componentWillUnmount() {
 		window.removeEventListener('resize', this.resize);
-		this.tty.destroy();
+
+		// Don't destroy tty on unmount if this Terminal is persistent
+		if (!this.props.persistent) {
+			this.tty.destroy();
+		}
 	}
 
+	// Explicitly calling '.destroy()' will always work, even with the 'persistent' property
 	destroy() {
 		window.removeEventListener('resize', this.resize);
 
-		// If the Terminal was not instantiated with an existing tty instance,
-		// destroy the instance when the component unmounts
-		if (!this.props.ttyInstance) {
-			this.tty.destroy();
-		}
+		this.tty.destroy();
```

Comment:
Should we also set it to null or something?

Reply:
@thgreasi Is this good practice for GC cycles? I would have thought that it would get ditched once the React component is removed

Reply:
Well, browsers can GC cyclic references, but this would make it a bit easier for them.